chichiwang/exercise-js
# Create a `pages/` directory to house application Pages

## Description
Linked to Issue: #19
Add a `pages/` directory to house application pages, plugged into the routing in `app/Routes`. Move `app/Home` to `pages/Home`. The rationale is that in the future there will top level directories that reveal composite pieces that comprise the software intent:
* `client/`: houses web client concerns, principally hydration of the application
* `server/`: houses web server concerns, principally the web server itself
* `app/`: houses application-specific concerns, including the basic high level markup and SPA routing mechanisms, as well as global hooks and utilities
* `pages/`: houses the web pages that will comprise the SPA
* `components/`: houses reusable components used throughout the web pages
* Other directories specific to the nature of the webpage itself (ex: exercises, products, articles, blog posts, etc)

## Changes
* Move `app/Home` to `pages/Home`
* Update config files to create a `pages` alias for the `pages/` directory:
  * `eslintrc.js`
  * `tsconfig.json`
  * `webpack-config/paths`
  * `webpack-config/base`
  * `jest.config.js`
* Add `pages/` dir to the `App Tests` project in `jest.config.js`

## Steps to QA
* Pull down and switch to this branch
* Run `npm run build:run` and ensure the assets build and server runs with no issue
  * Verify in browser
* Run `npm start` and ensure the assets build and server runs with no issue
  * Verify in browser
